web/datapipeline/telemetry_projector.go

```diff
@@ -0,0 +1,106 @@
+package datapipeline
+
+import (
+	"bytes"
+	"encoding/json"
+
+	log "github.com/sirupsen/logrus"
+	discoveryModels "github.com/trento-project/trento/agent/discovery/models"
+	"github.com/trento-project/trento/internal/cloud"
+	"github.com/trento-project/trento/internal/subscription"
+	"github.com/trento-project/trento/web/models"
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+func NewTelemetryProjector(db *gorm.DB) *projector {
+	telemetryProjector := NewProjector("telemetry", db)
+
+	telemetryProjector.AddHandler(HostDiscovery, telemetryProjector_HostDiscoveryHandler)
+	telemetryProjector.AddHandler(CloudDiscovery, telemetryProjector_CloudDiscoveryHandler)
+	telemetryProjector.AddHandler(SubscriptionDiscovery, telemetryProjector_SubscriptionDiscoveryHandler)
+
+	return telemetryProjector
+}
+
+func telemetryProjector_HostDiscoveryHandler(dataCollectedEvent *DataCollectedEvent, db *gorm.DB) error {
+	decoder := payloadDecoder(dataCollectedEvent.Payload)
+
+	var discoveredHost discoveryModels.DiscoveredHost
+	if err := decoder.Decode(&discoveredHost); err != nil {
+		log.Errorf("can't decode data: %s", err)
+		return err
+	}
+
+	telemetryReadModel := models.Telemetry{
+		AgentID:       dataCollectedEvent.AgentID,
+		HostName:      discoveredHost.HostName,
+		CPUCount:      discoveredHost.CPUCount,
+		SocketCount:   discoveredHost.SocketCount,
+		TotalMemoryMB: discoveredHost.TotalMemoryMB,
+	}
+
+	return storeTelemetry(db, telemetryReadModel,
+		"host_name",
```

Comment:
I understand that this names are generated by gorm, but maybe we could document or make this names explicit in the model?

Reply:
Do you mean expliciting at a model level the `gorm` annotation? That we can do.

Reply:
Yes, to have some sort of relation (at least when search in the code)

Reply:
Super! Done.